File: rxbinding/src/main/java/com/jakewharton/rxbinding2/view/ViewScrollChangeEvent.java
package com.jakewharton.rxbinding2.view;

import android.support.annotation.CheckResult;
import android.support.annotation.NonNull;
import android.view.View;
import com.google.auto.value.AutoValue;

/**
 * A scroll-change event on a view.
 * <p>
 * <strong>Warning:</strong> Instances keep a strong reference to the view. Operators that cache
 * instances have the potential to leak the associated {@link android.content.Context}.
 */
@AutoValue
public abstract class ViewScrollChangeEvent {
  @CheckResult @NonNull
  public static ViewScrollChangeEvent create(@NonNull View view, int scrollX, int scrollY,
      int oldScrollX, int oldScrollY) {
    return new AutoValue_ViewScrollChangeEvent(view, scrollX, scrollY, oldScrollX, oldScrollY);
  }

  ViewScrollChangeEvent() {
  }

  /** The view from which this event occurred. */
  @NonNull public abstract View view();
  public abstract int scrollX();
  public abstract int scrollY();
  public abstract int oldScrollX();
  public abstract int oldScrollY();
}

File: rxbinding/src/main/java/com/jakewharton/rxbinding2/view/ViewScrollChangeEvent.kt
package com.jakewharton.rxbinding2.view

import android.view.View
import androidx.annotation.CheckResult

/**
 * A scroll-change event on a view.
 *
 * **Warning:** Instances keep a strong reference to the view. Operators that cache
 * instances have the potential to leak the associated [android.content.Context].
 */
data class ViewScrollChangeEvent(
  /** The view from which this event occurred.  */
  val view: View,
  val scrollX: Int,
  val scrollY: Int,
  val oldScrollX: Int,
  val oldScrollY: Int
)
